src/braket/circuits/gates.py

```diff
@@ -597,18 +663,33 @@ def to_ir(self, target: QubitSet):
     def to_matrix(self) -> np.ndarray:
         return np.array([[1.0, 0.0], [0.0, np.exp(1j * self.angle)]], dtype=complex)
 
+    def bind_values(self, **kwargs):
+        """
+        Takes in parameters and attempts to assign them to values.
+
+        Args:
+            **kwargs: The parameters that are being assigned.
+
+        Returns:
+            Gate.PhaseShift: A new Gate of the same type with the requested
+            parameters bound.
+
+        """
+        new_angle = self.angle if str(self.angle) not in kwargs else kwargs[str(self.angle)]
```

Comment:
There are at least 9 repeats of this code. Can we put it in a superclass or helper function?

Reply:
Will add this in the next commit.